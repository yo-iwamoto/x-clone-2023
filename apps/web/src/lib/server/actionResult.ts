export const actionResult = {
  success: true as const,
  failure: false as const,
};
