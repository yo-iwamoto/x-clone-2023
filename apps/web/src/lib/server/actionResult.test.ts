import { actionResult } from './actionResult';

describe('actionResult', () => {
  it('snapshot unchanged', () => {
    expect(actionResult).toMatchSnapshot();
  });

  it('success が true 型であること', () => {
    expectTypeOf(actionResult.success).toMatchTypeOf<true>();
  });

  it('failure が false 型であること', () => {
    expectTypeOf(actionResult.failure).toMatchTypeOf<false>();
  });
});
