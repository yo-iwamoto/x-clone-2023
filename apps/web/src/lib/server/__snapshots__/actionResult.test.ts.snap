// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`actionResult > snapshot unchanged 1`] = `
{
  "failure": false,
  "success": true,
}
`;
